benchmark_runner/benchmark_operator/benchmark_operator_workloads.py

```diff
@@ -821,21 +822,29 @@ def run_workload(self, workload: str):
         self.make_deploy_benchmark_controller_manager(runner_path=environment_variables.environment_variables_dict['runner_path'])
 
         # Start collection of Prometheus snapshot
-        extract_prometheus_snapshot = self.__environment_variables_dict.get('extract_prometheus_snapshot', 'True')
-        if extract_prometheus_snapshot.lower() == 'true':
-
+        if self.__enable_prometheus_snapshot:
             artifacts_path = self.__environment_variables_dict.get('run_artifacts_path', '')
-            if not os.path.isdir(artifacts_path):
-                os.mkdir(artifacts_path)
-            snapshot = PrometheusSnapshot(oc=self.__oc, artifacts_path=artifacts_path, verbose=True)
-            snapshot.prepare_for_snapshot()
+            try:
+                if not os.path.isdir(artifacts_path):
+                    os.mkdir(artifacts_path)
+                snapshot = PrometheusSnapshot(oc=self.__oc, artifacts_path=artifacts_path, verbose=True)
+                snapshot.prepare_for_snapshot()
+            except PrometheusSnapshotError as err:
+                raise PrometheusSnapshotFailed(err)
+            except Exception as err:
+                raise PrometheusSnapshotFailed(err)
```

Comment:
I think it should be just err, we dont know if that relate to [Prometheus](https://github.com/redhat-performance/benchmark-runner/blob/8c7f55f8e2504e6cab1348659430497668bef238/benchmark_runner/benchmark_operator/benchmark_operator_workloads.py#L466)